content/en/blog/2020/custom-istio-dns-name/index.md

```diff
@@ -0,0 +1,125 @@
+
+{{< text yaml >}}
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: istiocontrolplane-default
+  namespace: istio-operator
+spec:
+  addonComponents:
+    istiocoredns:
+      enabled: true
+{{< /text >}}
+
+When this deploys, we have the `istiocoredns` component that has `coredns` installed as well as the `istio-coredns-plugin` which is used to read `ServiceEntry`s and serve DNS queries for their hostname with IP addresses. The interesting bit is how it's configured out of the box. The default out of the box config is in the `coredns` configmap:
+
+{{< text bash >}}
+$ kubectl get cm coredns -n istio-system
+{{< /text >}}
+
+Looking into that configmap, we see the following configuration for Istio's `coredns`:
+
+{{< text json >}}
+.:53 {
+      errors
+      health
+
+      # Removed support for the proxy plugin: https://coredns.io/2019/03/03/coredns-1.4.0-release/
+      grpc global 127.0.0.1:8053
+      forward . /etc/resolv.conf {
+        except global
+      }
+
+      prometheus :9153
+      cache 30
+      reload
+    }
+{{< /text >}}
+
+Understanding how [coredns plugins work](https://coredns.io/plugins/grpc/) is helpful here. We see with the `grpc` statement that we're forwarding `global` domains to `127.0.0.1:8053`, and we're forwarding everything else to `/etc/resolv.conf`. The `istio-coredns-plugin` plugin is another container co-deployed with the `coredns` server and is listening on that `127.0.0.1:8053` port within the pod. If we want to change or add domains that can resolve based on what `ServiceEntry`s are created, we need to modify this file.
+
+For example, to add a `region` domain, we could do something like this:
+
+{{< text yaml>}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: coredns
+  namespace: istio-system
+  labels:
+    app: istiocoredns
+    release: istio
+data:
+  Corefile: |
+    region {
+             grpc . 127.0.0.1:8053
+          }
+    .:53 {
+          errors
+          health
+          grpc global 127.0.0.1:8053
+          forward . /etc/resolv.conf {
+            except global
+          }
+          prometheus :9153
+          cache 30
+          reload
+        }
+{{< /text >}}
+
+With this configuration of Istio's `coredns` component, we could have `ServiceEntry`s that have either hostnames of `foo.bar.region` or `foo.bar.global` resolvable by this DNS server.
+
+## Enable stubs for kube-dns
+
+The last step is to add this new `region` domain stub to the `kube-dns` entries. This is covered in the [replicated control-plane docs](/docs/setup/install/multicluster/gateways/#setup-dns).
+
+## Next steps
+
+In this blog post we saw how to customize the domain name used in global service discovery and routing within Istio. There may be other usecases for customizing this DNS resolution behavior, so hopefully this is helpful for others. We encourage you to try out [Istio multi-cluster](/docs/setup/install/multicluster/gateways).
```

Comment:
```suggestion
This blog post explained how to customize the domain name used in global service discovery and routing within Istio. For more information about Istio multi-cluster deployment, see [Replicated control planes](/docs/setup/install/multicluster/gateways).
```

Reply:
I don't think you need the line about there being other use cases, nor the "hopefully this is helpful for others" part. But if you do end up keeping those parts, then change "usecases" to "use cases".